pkg/networkservice/common/connect/networkservice_client.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connect
+
+import (
+	"context"
+	"net/url"
+
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/clienturl"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/heal"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/adapters"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
+)
+
+// NewNetworkServiceClient - adapts connect server chain to client
+func NewNetworkServiceClient(ctx context.Context, connectTo *url.URL, dialOptions []grpc.DialOption, additionalFunctionality ...networkservice.NetworkServiceClient) networkservice.NetworkServiceClient {
```

Comment:
So I guess here we also can introduce option pattern for dial options, additional functionality as in registry connect:
https://github.com/networkservicemesh/sdk/blob/main/pkg/registry/common/connect/option.go
https://github.com/networkservicemesh/sdk/blob/main/pkg/registry/common/connect/nse_client.go#L46

Reply:
Done